src/api/Feed.js

```diff
@@ -0,0 +1,1022 @@
+/**
+ * @flow
+ * @file Helper for activity feed API's
+ * @author Box
+ */
+import type { $AxiosXHR } from 'axios';
+import uniqueId from 'lodash/uniqueId';
+import noop from 'lodash/noop';
+import omit from 'lodash/omit';
+import Base from './Base';
+import {
+    COMMENTS_FIELDS_TO_FETCH,
+    TASKS_FIELDS_TO_FETCH,
+    VERSIONS_FIELDS_TO_FETCH,
+    TASK_ASSIGNMENTS_FIELDS_TO_FETCH
+} from '../util/fields';
+import CommentsAPI from './Comments';
+import VersionsAPI from './Versions';
+import TasksAPI from './Tasks';
+import TaskAssignmentsAPI from './TaskAssignments';
+import { getBadItemError, getBadUserError } from '../util/error';
+import messages from '../components/messages';
+import { VERSION_UPLOAD_ACTION, VERSION_RESTORE_ACTION } from '../constants';
+
+const DEFAULT_START = 0;
+const DEFAULT_END = 1000;
+const TASK_INCOMPLETE = 'incomplete';
+const TASK_ASSIGNMENT_COLLECTION = 'task_assignment_collection';
+
+type FeedItemsCache = {
+    hasError: boolean,
+    items: FeedItems
+};
+
+class Feed extends Base {
+    /**
+     * @property {VersionsAPI}
+     */
+    versionsAPI: VersionsAPI;
+
+    /**
+     * @property {CommentsAPI}
+     */
+    commentsAPI: CommentsAPI;
+
+    /**
+     * @property {TasksAPI}
+     */
+    tasksAPI: TasksAPI;
+
+    /**
+     * @property {TaskAssignmentsAPI}
+     */
+    taskAssignmentsAPI: Array<TasksAPI | TaskAssignmentsAPI>;
+
+    /**
+     * @property {string}
+     */
+    id: string;
+
+    /**
+     * @property {boolean}
+     */
+    hasError: boolean;
+
+    constructor(options: Options) {
+        super(options);
+        this.taskAssignmentsAPI = [];
+    }
+
+    /**
+     * Creates a key for the cache
+     *
+     * @param {string} id folder id
+     * @return {string} key
+     */
+    getCacheKey(id: string): string {
+        return `feedItems_${id}`;
```

Comment:
Probably move this in constants.js with other keys

Reply:
done